src/Core/Client/Adapter/TimeoutAwareTrait.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Solarium\Core\Client\Adapter;
+
+/**
+ * @internal
+ */
+trait TimeoutAwareTrait
+{
+    /**
+     * @var int|null
+     */
+    private $timeout;
```

Comment:
in Solarium 6 we can initialize this with `TimeoutAwareInterface::DEFAULT_TIMEOUT`